resources/git-hooks/commit-msg

```diff
@@ -12,14 +12,17 @@
 # Reference: https://github.com/keymanapp/keyman/wiki/Pull-Request-and-Commit-workflow-notes
 #
 
-#
-# Configurable options for types, scopes and message length.
-#
-
-types=(fix feat chore change docs style refactor test)
-scopes=(android ci common developer ios linux mac web windows)
-min_length=4
-max_length=72
+# Step 1 - resolve the source directory of the hook, ALSO resolving any symlinks.
+# From https://stackoverflow.com/a/246128.
+SCRIPT_SOURCE="${BASH_SOURCE[0]}"
```

Comment:
I've got another standard pattern in the master branch which I think is slightly cleaner (relies on greadlink on mac but since we are using that anyway...):

```
## START STANDARD BUILD SCRIPT INCLUDE
# adjust relative paths as necessary
THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
. "$(dirname "$THIS_SCRIPT")/build-utils.sh"
## END STANDARD BUILD SCRIPT INCLUDE
```

Reply:
It's worth noting that the `cd` pattern can fail in some situations.